src/Illuminate/Support/Str.php

```diff
@@ -379,6 +379,25 @@ public static function deduplicate(string $string, string $character = ' ')
         return preg_replace('/'.preg_quote($character, '/').'+/u', $character, $string);
     }
 
+    /**
+     * Truncates a string to a specified character limit and appends an ellipsis if necessary.
+     *
+     * @param  string  $value
+     * @param  int  $limit
+     * @param  string  $ellipsis
+     * @return string
+     */
+    public static function ellipsis(string $value, int $limit, string $ellipsis = '...'): string
+    {
+        if (mb_strlen($value) <= $limit) {
+            return $value;
+        }
+
+        $cut = $limit - mb_strlen($ellipsis);
```

Comment:
Imagine, the string is between the limit and the ellipsis length. What should happen?
* nothing (_proposed implementation_)
* an exception shall be thrown

```suggestion
        $ellipsisLength = mb_strlen($ellipsis);
        $originalLength = mb_strlen($value)
        $cut = $limit - $ellipsisLength;
        
        if ($originalLength <= $ellipsisLength) {
            throw new InvalidArgumentException('String must be at least ' . ($cut + 1) . ' characters long, ' . ' $originalLength . ' given');
        }
```